MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm

```diff
@@ -615,47 +628,60 @@ - (void)startCrashProcessing {
 }
 
 - (void)processCrashReports {
+  
+  // Handle 'disabled' state all at once to simplify the logic that follows.
+  if (!self.isEnabled) {
+    MSLogDebug([MSCrashes logTag], @"Crashes service is disabled; discard all crash reports");
+    [self deleteAllFromCrashesDirectory];
+    [MSWrapperExceptionManager deleteAllWrapperExceptions];
+    return;
+  }
+
+  // Must now be in 'enabled' state; start crash processing for real.
```

Comment:
The reason why I brought up this feedback is that if you assume the state of Crashes is always "enabled", it will be dangerous because the state can be changed in another thread. I don't see any specific code blocks that assume the state enabled, it should be fine. You probably need to revise the comment not to say "Must now be in 'enabled' state".